lti_consumer/lti_xblock.py

```diff
@@ -849,6 +849,23 @@ def get_lti_1p1_user_id(self):
 
         return self.anonymous_user_id
 
+    def get_lti_1p1_user_from_user_id(self, user_id):
+        """
+        Returns the user object associated with a user_id. This is used in LTI 1.1/2.0 integrations for calls to the
+        LTI 1.1 Basic Outcomes service and the LTI 2.0 Results service. Tool Providers may make calls to this library's
+        endpoints with a user identifier. This function returns a user object associated with that user identifier.
+
+        The user identifier may be a course-anonymized user ID (i.e. the anonymous_user_id) or the global, consistent
+        user ID (i.e. the external_user_id). This functions returns the correct User object.
+        """
+        if external_user_id_1p1_launches_enabled(self.scope_ids.usage_id.context_key):
```

Comment:
I _think_ this is right, based on https://github.com/openedx/xblock-lti-consumer/pull/249.

Reply:
yeah that looks like the sort of replacement done there: https://github.com/openedx/xblock-lti-consumer/pull/249/files#diff-6e408862179502a0ac809fab914e028b18311a17a3b0874469bd6416bc4a449fL73

Reply:
since the IDs are different formats we could detect that difference, but this seems cleaner